import pytest
from app.restore_names import restore_names
from typing import Callable


@pytest.fixture()
def incorrect_users_list() -> None:
    users = [
        {
            "first_name": None,
            "last_name": "Holy",
            "full_name": "Jack Holy",
        },
        {
            "last_name": "Adams",
            "full_name": "Mike Adams",
        }
    ]
    return users


def test_restore_names_adds_first_name(
        incorrect_users_list: Callable
) -> None:
    users = incorrect_users_list
    restore_names(users)
    assert users == [
        {
            "first_name": "Jack",
            "last_name": "Holy",
            "full_name": "Jack Holy",
        },
        {
            "first_name": "Mike",
            "last_name": "Adams",
            "full_name": "Mike Adams",
        }
    ]
